src/components/flyout/flyout.spec.tsx

```diff
@@ -76,6 +77,37 @@ describe('EuiFlyout', () => {
         expect(cy.get('[data-test-subj="flyoutSpec"]').should('not.exist'));
       });
     });
+  });
+
+  describe('Close behavior: outside clicks', () => {
+    const FlyoutWithToasts = ({ children = childrenDefault, ...rest }) => {
```

Comment:
It might be useful to add a comment here on "why toasts?" for context for people who come to this code after this bugfix/PR
```suggestion
    // We're using toasts here to trigger outside clicks, as a UX case where
    // we would generally expect toasts overlaid on top of a flyout *not* to close the flyout
    const FlyoutWithToasts = ({ children = childrenDefault, ...rest }) => {
```